src/SphericalCoordinates.cc

```diff
@@ -219,6 +313,63 @@ void SphericalCoordinates::SetSurface(const SurfaceType &_type)
   }
 }
 
+//////////////////////////////////////////////////
+void SphericalCoordinates::SetSurface(
+    const SurfaceType &_type,
+    const double _axisEquatorial,
+    const double _axisPolar,
+    const double _flattening)
+{
+  if ((_type != EARTH_WGS84) &&
+      (_type != MOON_SCS) &&
+      (_type != CUSTOM_SURFACE))
+  {
+    std::cerr << "Unknown surface type["
+      << _type << "]\n";
+    return;
+  }
+
+  this->dataPtr->surfaceType = _type;
+  if (_axisEquatorial > 0)
+  {
+    this->dataPtr->ellA = _axisEquatorial;
+  }
+  else
+  {
+    std::cerr << "Value of _axisEquatorial should be greater than zero "
+      " defaulting to Earth's equatorial radius." << std::endl;
+    this->dataPtr->ellA = g_EarthWGS84AxisEquatorial;
+  }
+
+  if (_axisPolar > 0)
+  {
+    this->dataPtr->ellB = _axisPolar;
+  }
+  else
+  {
+    std::cerr << "Value of _axisPolar should be greater than zero "
+      " defaulting to Earth's polar radius." << std::endl;
+    this->dataPtr->ellB = g_EarthWGS84AxisPolar;
+  }
+
+  if (_flattening >= 0)
+  {
+    this->dataPtr->ellF = _flattening;
+  }
+  else
+  {
+    std::cerr << "Value of _flattening should be greater than zero "
+      " defaulting to Earth's flattening value."<< std::endl;
```

Comment:
Just checking, the condition has `_flattening >= 0`, should the error message say `.. greater or equal to zero..`?

Reply:
Done here : https://github.com/gazebosim/gz-math/pull/434/commits/b1baee53f003c614bfeec7c28afb2458a82cd7bc